### PR TITLE
926/protocol handlers

### PR DESCRIPTION
**Pull request by sushi2k**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to Mitre/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectivnes of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

#926

### CODE REVIEW

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 1]
```diff
@@ -63,6 +63,65 @@ URI => "myapp://testpayload'"
...
+```
+
+Swift:
+```
+webView.configuration.preferences.setValue(true, forKey: "allowFileAccessFromFileURLs")
+```
+
+By default WKWebView disables file access. If one or more of the above methods is/are activated, you should determine whether the method(s) is/are really necessary for the app to work properly.
+
+> Please also verify which WebView class is used. WKWebView should be used nowadays, as `UIWebView` is deprecated.
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: The > could better just be part of the paragraph above.

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 2]
```diff
@@ -63,6 +63,65 @@ URI => "myapp://testpayload'"
...
+
+The URL specified in `loadFileURL` should be checked for dynamic parameters that can be manipulated; their manipulation may lead to local file inclusion.
+
+Use the following best practices as defensive-in-depth measures:
+- Create a whitelist that defines local and remote web pages and protocols that are allowed to be loaded.
+- Create checksums of the local HTML/JavaScript files and check them while the app is starting up. Minify JavaScript files to make them harder to read.
+
+#### Dynamic Analysis
+
+To identify the usage of protocol handlers, look for ways to trigger phone calls and ways to access files from the file system while you're using the app.
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: This is still a bit vague: can you elaborate on it a bit more?

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 3]
```diff
@@ -63,6 +63,70 @@ URI => "myapp://testpayload'"
...
+```objc
+[webView.configuration.preferences setValue:@YES forKey:@"allowFileAccessFromFileURLs"];
+```
+
+Swift:
+```swift
+webView.configuration.preferences.setValue(true, forKey: "allowFileAccessFromFileURLs")
+```
+
+> By default WKWebView disables file access. If one or more of the above methods is/are activated, you should determine whether the method(s) is/are really necessary for the app to work properly.
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: why is there a > here?

Comment (commjoen) [reply Document/0x06h-Testing-Platform-Interaction.md]: Please remove that and then we can merge :)

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 4]
```diff
@@ -73,11 +137,9 @@ iOS WebViews support JavaScript execution by default, so script injection and cr
 
 Besides potential script injection, there's another fundamental WebViews security issue: the WebKit libraries packaged with iOS don't get updated out-of-band like the Safari web browser. Therefore, newly discovered WebKit vulnerabilities remain exploitable until the next full iOS update [#THIEL].
 
-WebViews support different URL schemas, like for example tel. Detection of the [tel:// schema can be disabled](https://developer.apple.com/library/content/featuredarticles/iPhoneURLScheme_Reference/PhoneLinks/PhoneLinks.html "Phone Links on iOS") in the HTML page and will then not be interpreted by the WebView.
-
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: why is  this reference removed? Because it is very explanatory

Comment (sushi2k) [reply Document/0x06h-Testing-Platform-Interaction.md]: It wasn't removed, this one was moved to the test case I created, as it fit's better.

**Pull request merged**